lib/backend/firestore/firestorebk.go

```diff
@@ -549,41 +575,49 @@ func (b *Backend) CompareAndSwap(ctx context.Context, expected backend.Item, rep
 		return nil, trace.BadParameter("expected and replaceWith keys should match")
 	}
 
-	// The operations performed are get, compare values, and conditional update instead of using
-	// a transaction because the firestore api does not return the commit time from RunTransaction.
-	// See https://github.com/googleapis/google-cloud-go/issues/6841.
-	for i := 0; i < 3; i++ {
-		docSnap, err := b.svc.Collection(b.CollectionName).
-			Doc(b.keyToDocumentID(expected.Key)).
-			Get(ctx)
+	replaceWith.Revision = createRevisionV2()
+	replaceRec := newRecord(replaceWith, b.clock)
+
+	docRef := b.svc.Collection(b.CollectionName).Doc(b.keyToDocumentID(replaceWith.Key))
+
+	err := b.svc.RunTransaction(ctx, func(ctx context.Context, tx *firestore.Transaction) error {
+		docSnap, err := tx.Get(docRef)
 		if err != nil {
 			err = ConvertGRPCError(err)
```

Comment:
Should we return unexpected firestore errors verbatim to RunTransaction? We seem to check for status code not found in ConditionalUpdate and ConditionalDelete.

Reply:
Performing error conversion inside of `RunTransaction` is safe (the firestore client treats all errors returned by the callback as opaque), and I think its more readable to perform error handling at the location we first observe the error.  The only code we check for outside of `RunTransaction` is the `Aborted` code, which is returned by `RunTrasnaction` itself, not by one of the specific methods within the callback.